File: app/src/main/java/com/jann_luellmann/thekenapp/data/view_model/relationship/EventWithDrinksViewModel.java
package com.jann_luellmann.thekenapp.data.view_model.relationship;

import com.jann_luellmann.thekenapp.data.model.Drink;
import com.jann_luellmann.thekenapp.data.model.Event;
import com.jann_luellmann.thekenapp.data.model.EventDrinkCrossRef;
import com.jann_luellmann.thekenapp.data.model.relationship.EventWithDrinks;
import com.jann_luellmann.thekenapp.data.view_model.BaseViewModel;

import java.util.List;

import androidx.lifecycle.LiveData;

public class EventWithDrinksViewModel extends BaseViewModel<EventWithDrinks> {

    public EventWithDrinksViewModel() {
        super();
    }

    public LiveData<List<EventWithDrinks>> findAll() {
        return db.eventWithDrinksDAO().getAll();
    }

    public LiveData<EventWithDrinks> findByName(String name) {
        return db.eventWithDrinksDAO().findByName(name);
    }

    public LiveData<EventWithDrinks> findByEvent(Event event) {
        return db.eventWithDrinksDAO().findByName(event.getName());
    }

    @Override
    public void insert(long eventId, EventWithDrinks eventWithDrinks) {
        //nop
    }

    @Override
    public void delete(EventWithDrinks eventWithDrinks) {
        //nop
    }

    public void delete(long eventId, Drink drink) {
        executor.execute(() -> db.eventDrinkCrossDAO().delete(new EventDrinkCrossRef(eventId, drink.getDrinkId())));
    }
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/view_model/relationship/EventWithDrinksViewModel.kt
package com.jann_luellmann.thekenapp.data.view_model.relationship

import androidx.lifecycle.LiveData
import com.jann_luellmann.thekenapp.data.model.Drink
import com.jann_luellmann.thekenapp.data.model.Event
import com.jann_luellmann.thekenapp.data.model.EventDrinkCrossRef
import com.jann_luellmann.thekenapp.data.model.relationship.EventWithDrinks
import com.jann_luellmann.thekenapp.data.view_model.BaseViewModel

class EventWithDrinksViewModel : BaseViewModel<EventWithDrinks?>() {
    override fun findAll(): LiveData<List<EventWithDrinks?>>? {
        return db.eventWithDrinksDAO().all
    }

    fun findByName(name: String?): LiveData<EventWithDrinks?>? {
        return db.eventWithDrinksDAO().findByName(name)
    }

    fun findByEvent(event: Event): LiveData<EventWithDrinks?>? {
        return db.eventWithDrinksDAO().findByName(event.name)
    }

    override fun insert(eventId: Long, eventWithDrinks: EventWithDrinks?) {
        //nop
    }

    override fun delete(eventWithDrinks: EventWithDrinks?) {
        //nop
    }

    fun delete(eventId: Long, drink: Drink) {
        executor.execute {
            db.eventDrinkCrossDAO().delete(EventDrinkCrossRef(eventId, drink.drinkId))
        }
    }
}